viral/two_photon.py

```diff
@@ -254,33 +256,236 @@ def place_cells(session: Cached2pSession, dff: np.ndarray) -> None:
     )
 
 
-if __name__ == "__main__":
+def activity_trial_speed(
+    trial: TrialInfo, dff: np.ndarray, wheel_circumference: float, verbose: bool = False
+) -> np.ndarray | None:
+    position = degrees_to_cm(
+        np.array(trial.rotary_encoder_position), wheel_circumference
+    )
 
-    mouse = "JB018"
-    date = "2024-12-06"
+    frame_position = np.array(
+        [
+            state.closest_frame_start
+            for state in trial.states_info
+            if state.name in ["trigger_panda", "trigger_panda_post_reward"]
+        ]
+    )
+
+    assert len(position) == len(frame_position)
 
-    with open(HERE.parent / "data" / "cached_2p" / f"{mouse}_{date}.json", "r") as f:
-        session = Cached2pSession.model_validate_json(f.read())
-    print(f"Total number of trials: {len(session.trials)}")
-    print(
-        f"number of trials imaged {len([trial for trial in session.trials if trial_is_imaged(trial)])}"
+    # TODO: Think about step_size! Has to evenly divide the range, however we had it at 30 before (0-150; 150-180; ...)
+    first_position = 10
+    last_position = 170
+    step_size = 1
```

Comment:
Have you plotted this with step_size = 1? My guess would be that there'll be a large variance but I could be wrong